packaging/version.py

```diff
@@ -198,7 +198,12 @@ class Version(_BaseVersion):
 
     def __init__(self, version):
         # Validate the version and parse it into pieces
-        match = self._regex.search(version)
+        try:
+            match = self._regex.search(version)
+        except TypeError:
+            raise InvalidVersion(
+                "Invalid type: '{0}' is not a string".format(type(version))
```

Comment:
@xavfernandez I agree.
I'll try to make a short, clear and nice message with all that info, but I'm struggling a bit with that (english is not my native language, and I end up having some weird ways to say stuff).